micrometer-core/src/main/java/io/micrometer/core/instrument/binder/mongodb/MongoCommandUtil.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2019 VMware, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.mongodb;
+
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import io.micrometer.core.instrument.util.StringUtils;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+
+class MongoCommandUtil {
+
+    // See https://docs.mongodb.com/manual/reference/command for the command reference
+    private static final Set<String> COMMANDS_WITH_COLLECTION_NAME = new LinkedHashSet<>(Arrays.asList(
+            "aggregate", "count", "distinct", "mapReduce", "geoSearch", "delete", "find", "findAndModify",
+            "insert", "update", "collMod", "compact", "convertToCapped", "create", "createIndexes", "drop",
+            "dropIndexes", "killCursors", "listIndexes", "reIndex"));
+
+    /**
+     * Attempts to determine the name of the collection a command is operating on.
+     *
+     * <p>Because some commands either do not have collection info or it is problematic to determine the collection info,
+     * there is an allow list of command names {@link #COMMANDS_WITH_COLLECTION_NAME} used. If {@code commandName} is
+     * not in the allow list or there is no collection info in {@code command}, it will use the content of the
+     * {@code 'collection'} field on {@code command}, if it exists.
+     *
+     * <p>Taken from <a href="https://github.com/openzipkin/brave/blob/master/instrumentation/mongodb/src/main/java/brave/mongodb/TraceMongoCommandListener.java#L115>TraceMongoCommandListener.java</a>
+     *
+     * @param commandName name of the mongo command
+     * @param command mongo command object
+     * @return optional collection name or empty if could not be determined or not in the allow list of command names
+     */
+    Optional<String> determineCollectionName(final String commandName, final BsonDocument command) {
+        if (COMMANDS_WITH_COLLECTION_NAME.contains(commandName)) {
+            final Optional<String> collectionName = getNonEmptyBsonString(command.get(commandName));
+            if (collectionName.isPresent()) {
+                return collectionName;
+            }
```

Comment:
Agreed. At the very least I typically use `orElse` or `orElseGet` in this scenario. At this point I am more than certain the code was chopped/refactored at some point and I did not do another pass over it. 

Good suggestions.

Reply:
Ok, I don't feel so crazy now. I have the code like this for a reason...

We only want to go through the `getNonEmptyBsonString(command.get(commandName))` when the command is in the allow list (other commands do not hold the collection info under the name of the command in the bson map. If that is not present then we want to do the default 'collection' lookup. 


```java
COMMANDS_WITH_COLLECTION_NAME.contains(commandName))

Reply:
I might be missing something but that is still the case with the suggested change. Let me try to walk trough this.
The original code:
```java
if (COMMANDS_WITH_COLLECTION_NAME.contains(commandName)) {
    Optional<String> collectionName = getNonEmptyBsonString(command.get(commandName));
    if (collectionName.isPresent()) {
        return collectionName;
    }
}

return getNonEmptyBsonString(command.get("collection"));
```
This has three branches:
1. Command is in the allow list and non-empty optional -> `getNonEmptyBsonString(command.get(commandName))`
2. Command is in the allow list and empty optional -> `getNonEmptyBsonString(command.get("collection"))`
3. Command is not in the allow list -> `getNonEmptyBsonString(command.get("collection"))` `//same as #2`

So if we do this:
```java
if (COMMANDS_WITH_COLLECTION_NAME.contains(commandName)) {
    return getNonEmptyBsonString(command.get(commandName))
            .orElseGet(() -> getNonEmptyBsonString(command.get("collection")));
}

return getNonEmptyBsonString(command.get("collection"));
```

We should get the same behavior, right?
(Please notice that I only replaced the inner `if`, the outer allow list check is still there as-is).

Reply:
Yep, I understood your suggestion. What I was trying to convey is that the suggested change is more complex as there is now repeated call to `getNonEmptyBsonString(command.get("collection")))` in an `orElseGet()`. This will currenlty get handled when it falls through to the default `return getNonEmptyBsonString(command.get("collection"));`. 

In this case, IMO the `Optional.isPresent` is the cleaner of the two options as it does not violate DRY and it easier to read. 

All of that being said, I don't have a super strong opinion and will go w/ your suggestion if need be. Just lmk.

Reply:
To me, calling `ifPresent` (and `get`) on an `Optional` is usually a bad smell. I think having `getNonEmptyBsonString` once in a supplier once in a normal method call is not a big sacrifice but I get your point,  let's keep it as is (we can also put hat call behind a supplier but that would unnecessarily complicate it).